### DISCUSSION TITLE
503 Service Temporarily Unavailable

### DISCUSSION THREAD

**Discussion by SinceTheCross** [fastlane/fastlane#21178] [Q&A]
[14:34:07]: Starting login with user ''
Available session is not valid any more. Continuing with normal login.
Unable to find 'X-Apple-HC-Bits' and 'X-Apple-HC-Challenge' to make hashcash

Looking for related GitHub issues on fastlane/fastlane...

➡️  Reading keychain entry, because either user or password were empty
    https://github.com/fastlane/fastlane/issues/21119 [open] 1 💬
    2 weeks ago

➡️  Action to inject a specific properties file to override one in an IPA?
    https://github.com/fastlane/fastlane/issues/503 [closed] 3 💬
    26 Sep 2016

🔗  You can ⌘ + double-click on links to open them directly in your browser.

[!] The request could not be completed because:
	<html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body>
<center><h1>503 Service Temporarily Unavailable</h1></center>
<hr><center>Apple</center>
</body>
</html>

After executing fastlane match appstore, this error was returned, but the system certificate has been successfully imported

**Comment by superClown** [comment]
I have also encountered the same problem.

**Comment by prnewman** [comment]
This is happening to me as well. It seems that the session expires before the provisioning profiles are generated, and fastlane is unable to renew the session. I'm on the latest version of fastlane (2.214.0).

**Comment by taschik** [comment]
Did anyone find a solution or workaround for this?

**Comment by Eddie-064** [comment]
I encountered the same issue as well today. Initially, I suspected that it might be due to too many attempts, and some online resources suggest to wait 24 hours and retry it. However, I skip this error now after changing some code. Specifically, I made changes to line 497 and 498 in `/opt/homebrew/Cellar/fastlane/2.214.0/libexec/gems/fastlane-2.216.0/spaceship/lib/spaceship/client.rb`. Here's what I modified:
```diff
-       #hashcash = nil
-       hashcash = self.fetch_hashcash
+       hashcash = nil
+       #hashcash = self.fetch_hashcash
```
This adjustment seems to have resolved the issue for me,  but I have no idea about the underlying mechanics.
If anyone can tell me why, it would be appreciated.

**Reply by Jay-57blocks** [reply]
Thank you, I encountered the same problem and solved it successfully, I don't know why the official has not fixed this problem yet

**Reply by theonetheycallneo** [reply]
We are experiencing the same error.

it appears to be that the tokens are expiring before the lane finishes execution.

perhaps there’s a way to set the duration of the session token?

for now we are using the json keys.

**Comment by x0000ff** [comment]
It looks like it was fixed in #21073. I had the same problem on `2.219.0` but after updating to `2.222.0` the problem's gone.

**Comment by timmkent** [comment]
This problem just popped up again on Oct 18, 2024.
Just a temporary problem or has Apple changed something?

Fastlane Version: 2.224.0
E.g. happens in get_certificates

```
[08:41:18]: Called from Fastfile at line 128
[08:41:18]: ```
[08:41:18]:     126:
[08:41:18]:     127:        # get certificates & profiles
[08:41:18]:  => 128:        get_certificates 
[08:41:18]:     129:        get_provisioning_profile
[08:41:18]:     130:
[08:41:18]: ```
[08:41:18]: <html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body>
<center><h1>503 Service Temporarily Unavailable</h1></center>
<hr><center>Apple</center>
</body>
</html>

```

**Comment by Lakston** [comment]
This is happening to us too, I tried twice yesterday, and this morning also, same error.

**Comment by davidmdt21** [comment]
yes same error for us ! please help if anybody fix this problem
Exception type: Spaceship::Tunes::Error

Looking for related GitHub issues on fastlane/fastlane...


[!] The request could not be completed because:
	<html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body>
<center><h1>503 Service Temporarily Unavailable</h1></center>
<hr><center>Apple</center>
</body>
</html>

should i use appstore connect api for authentication to apple developer ?

**Comment by andersonbarutti** [comment]
We're experiencing the same issue here as well. The problem started last night (October 18, 2024) and has persisted through today (October 19, 2024). We've made multiple attempts both yesterday evening and this morning, but the error continues to occur.
Some additional information that might be helpful:

1. We're using **Fastlane** version **2.224.0**.
2. The Apple Developer services status page shows all systems as operational, which makes this issue more puzzling.
3. The error occurs specifically during the sync_code_signing step, when **Fastlane** attempts to verify certificates and provisioning profiles on the Apple Developer Portal.
4. We've tried cleaning and reinstalling **Fastlane**, but the problem persists.
5. The error message is consistent:
  ```
  [!] The request could not be completed because:
     <html>
      <head><title>503 Service Temporarily Unavailable</title></head>
      <body>
        <center><h1>503 Service Temporarily Unavailable</h1></center>
        <hr><center>Apple</center>
      </body>
     </html>
   ```
6. We've attempted the process from different networks to rule out local network issues.

It seems that this might be a widespread issue affecting Apple's authentication or certificate verification systems, despite their status page indicating no problems. Has anyone found a workaround or received any official communication from Apple about this?
We're considering trying the workaround suggested by Eddie-064 (modifying the spaceship/client.rb file), but we're hesitant to make changes to the Fastlane source code. Any insights or alternative solutions would be greatly appreciated.

**Reply by davidmdt21** [reply]
thank for helpfully reply . please let us know here ,when you  tested some thing suggested by Eddie-064 . 
also i think we should make changes own codes for use app store connect api  https://docs.fastlane.tools/app-store-connect-api/
what you think man ?

**Reply by andersonbarutti** [reply]
Hi @davidmdt21 ,

Thank you for your response. I appreciate your interest in potential solutions.

Regarding Eddie-064's suggestion:
We haven't tried modifying the `spaceship/client.rb` file yet. While it seems to have worked for some, we're cautious about altering Fastlane's core files as it could lead to unforeseen issues and complicate future updates. However, if the problem persists, we might consider it as a last resort.

As for using the App Store Connect API:
You raise an interesting point about transitioning to the App Store Connect API. Here are some thoughts:

1. **Potential Benefits:**
   - More direct communication with Apple's services, possibly bypassing some of the issues we're facing with Fastlane's built-in methods.
   - Greater control over the authentication and submission process.
   - Possibly more stable in the long run, as it's Apple's official API.

2. **Considerations:**
   - It would require significant changes to our current workflow and CI/CD pipelines.
   - There might be a learning curve and initial setup time.
   - We'd need to ensure it integrates well with our existing Fastlane setup for other tasks.

3. **Current Situation:**
   - Given that this seems to be a widespread issue affecting many Fastlane users, it might be a temporary problem on Apple's end.
   - Before making such a significant change, it might be worth waiting a bit longer to see if the issue resolves or if Fastlane releases an update addressing it.

4. **Next Steps:**
   - We'll continue monitoring the situation and the Fastlane GitHub issues for any official updates or widespread solutions.
   - If the problem persists for several more days, we might start looking into implementing the App Store Connect API as a more robust long-term solution.

**Comment by jieey1140** [comment]
same isssue..  Has anyone resolved this issue?

**Reply by Fantome5** [reply]
Same here

**Reply by torca2011** [reply]
The same problem, any know more information?

**Reply by standinga** [reply]
I have the same issue.

**Comment by felipecosta91** [comment]
I have the same issue. [2] :(

**Comment by MayankStx** [comment]
Same issues as everyone facing on Version `fastlane 2.224.0`

```
[!] The request could not be completed because:
<html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body>
<center><h1>503 Service Temporarily Unavailable</h1></center>
<hr><center>Apple</center>
</body>
</html>
```


Please fix this asap, My certificates are expired and can't renew them

<img width="752" alt="Screenshot 2024-10-20 at 10 31 58" src="https://github.com/user-attachments/assets/25820304-293d-4215-8a74-5704ac160e2e">

**Comment by rgagne** [comment]
Same issue here. I was on an older version of Fastlane (215 I think) and upgraded to 2.224 hoping that would resolve things but same problem. Not specific to version 224 but seems a general issue. 

Apple's system status page is showing everything up https://www.apple.com/support/systemstatus/ so possibly some change to the way the system authenticates.

**Comment by jinnnnn-n** [comment]
I've been facing this issue since yesterday and struggled a lot, but downgrading from Fastlane `2.224.0` to `2.219.0` has resolved the issue for now. It's just a temporary workaround but hope it helps.

Update Gemfile and run `bundle install`
```
gem "fastlane", "2.219.0"
````

**Comment by RoiLaredo** [comment]
same issue.  could no find any info from apple side.

**Comment by Fantome5** [comment]
Watch here: https://github.com/fastlane/fastlane/issues/26368

**Reply by danicunhac** [reply]
Thank you @Fantome5, the solution provided in that issue solved the problem.

**Comment by danicunhac** [comment]
Same here, tried upgrading fastlane and reinstalling spaceship, but nothing helped. Will try downgrading fastlane to `2.219.0` and if that doesn't work, next step is to modify the fastlane core file

**Comment by danicunhac** [comment]
Link to solution: https://github.com/fastlane/fastlane/issues/26368#issuecomment-2424190032

**Reply by Lakston** [reply]
upgrading to 2.225.0 solves the problem, thanks to everyone involved 🙏

**Comment by MayankStx** [comment]
Updating it to 2.225.0 only via brew solved my problems. There is a lot of discussion to update fastlane via GemFile or brew.
